### PR TITLE
Move Concat Linear out of Optimize Numerics

### PR DESCRIPTION
**Pull request by Gamrix**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67792
* #67201
* __->__ #67196

Differential Revision: [D32154788](https://our.internmc.facebook.com/intern/diff/D32154788)

### CODE REVIEW

**Review by eellison — COMMENTED** [review]
can you just remove https://github.com/pytorch/pytorch/blob/master/torch/jit/_freeze.py#L162 and have it call into `OptimizeFrozenGraph` ? originally there was some thought of python hackability etc, but we should just remove the duplication.

can you also hook up the pretranspose pass?

**Review by eellison — APPROVED** [review]

**Review by eellison — CHANGES_REQUESTED** [review]
we still need to replace the python api to call into this version, that's why no tests are breaking

**Review by eellison — COMMENTED** [review]
we're probably going to need to remove the MKLDNN Linear tests, that's fine. can we remove https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp#L939 the linear here and say that we get more speedup by pretransposing linear layers than running them in mkldnn

**Review by eellison — COMMENTED** [review]
hey, some of the tests need to be rewritten with conv

**Review by eellison — APPROVED** [review]
nice!!

**Comment by eellison** [comment]
hey - we have an internal user who's optimization is blocked by this. could we try to land it ?

File: torch/jit/_freeze.py [hunk 1]
```diff
@@ -121,8 +121,10 @@ def forward(self, input):
 def run_frozen_optimizations(mod, optimize_numerics: bool = True):
     r"""
     Runs a series of optimizations looking for patterns that occur in frozen graphs.
-    The current set of optimizations is:
+    The current set of optimizations includes:
         - Dropout Removal
+        - Pretranspose Linear Layers
```
Comment (Gamrix) [inline torch/jit/_freeze.py:126]: Remove this one

**Comment by Gamrix** [comment]
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32154788).

**Pull request closed without merging**